server/etcdserver/server.go

```diff
@@ -109,6 +109,7 @@ const (
 	readyPercentThreshold = 0.9
 
 	DowngradeEnabledPath = "/downgrade/enabled"
+	memorySnapshotCount  = 100
```

Comment:
Should we add a config item (i.e. `--compact-raft-log-interval-indices`) for this?  We don't have a feature flag for this PR. If users set the the same value as `--snapshot-count`, then it's similar to disable this feature.

Reply:
I think it's good to discuss whether a configuration option would be useful for user, however we cannot expose each numeric const as option, we don't want etcd options to feel like atomic plant control panel. 

For each option we should configure whether and how user will use it and whether it makes sense to support it long term. In this case I don't see reason to expose it as there is no tradeoff happening here.  It's not like snapshot count, which I expect dates to etcd v2 where you trade off performance vs time to recovery. Here it's always performance as time to recovery will be based on last snapshot.

Reply:
Actually I am not too much concerned about this option. I was thinking memory saving vs CPU increasing/throughput reducing might be a tradeoff, but based on your test result, it seems that there isn't too much impact on CPU & throughput.

The main point is that we changed the internal behaviour of etcdserver, but there is no way to disable/revert the behaviour in case of any regression.

Anyway, I am OK to keep it as it's for now.

Reply:
> The main point is that we changed the internal behaviour of etcdserver, but there is no way to disable/revert the behaviour in case of any regression.

For such cases we should consider using feature gates, the downside is that it will require keeping old code for multiple releases. I would consider taking an exception for v2storage clenaup, which I consider this change as it based on removal of snapshots. 

Post v3.7 we might consider feature gating such changes.

Reply:
> I would consider taking an exception for v2storage clenaup, which I consider this change as it based on removal of snapshots.

OK.